imag-todo/src/main.rs

```diff
@@ -16,210 +16,139 @@ extern crate libimagtodo;
 use std::process::exit;
 use std::process::{Command, Stdio};
 use std::io::stdin;
-use std::io::BufRead;
 
-use task_hookrs::import::{import_task, import_tasks};
+use toml::Value;
+
+use task_hookrs::import::import_tasks;
+use task_hookrs::status::TaskStatus;
 
 use libimagrt::runtime::Runtime;
-use libimagtodo::task::IntoTask;
+use libimagrt::setup::generate_runtime_setup;
+use libimagtodo::task::Task;
 use libimagerror::trace::trace_error;
 
 mod ui;
 
 use ui::build_ui;
 fn main() {
+    let rt = generate_runtime_setup("imag-todo",
+                                    &version!()[..],
+                                    "Interface with taskwarrior",
+                                    build_ui);
 
-    let name = "imag-todo";
-    let version = &version!()[..];
-    let about = "Interface with taskwarrior";
-    let ui = build_ui(Runtime::get_default_cli_builder(name, version, about));
-
-    let rt = {
-        let rt = Runtime::new(ui);
-        if rt.is_ok() {
-            rt.unwrap()
-        } else {
-            println!("Could not set up Runtime");
-            println!("{:?}", rt.unwrap_err());
-            exit(1);
-        }
-    };
+    match rt.cli().subcommand_name() {
+        Some("tw-hook") => tw_hook(&rt),
+        Some("list") => list(&rt),
+        _ => unreachable!(),
+    } // end match scmd
+} // end main
 
+fn tw_hook(rt: &Runtime) {
+    let subcmd = rt.cli().subcommand_matches("tw-hook").unwrap();
+    if subcmd.is_present("add") {
+        let stdin = stdin();
+        let stdin = stdin.lock();
 
+        match Task::import(rt.store(), stdin) {
+            Ok((_, uuid)) => info!("Task {} stored in imag", uuid),
```

Comment:
I dont think taskwarrior will like this output. As for the on-add hook, taskwarrior requires this output:
1.: A json-formatted task-object, because the add-hook might have changed some things (imag doesn't, but the object is still required)
2. (optional): A comment, which I think you are currently printing with `info!()`. I don't know the info-macro yet, so you should make sure it really gets printed to stdout, and not some logfile.